### ISSUE TITLE
Unable to upload app privacy details using fastlane

### ISSUE THREAD

**Issue by ceylino** [fastlane/fastlane#18658]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

We have been using the upload_app_privacy_details_to_app_store call to update our app privacy details since it was implemented in December (https://docs.fastlane.tools/uploading-app-privacy-details). It was working as expected until yesterday, when we noticed that the app privacy details were not being updated on the AppStoreConnect side and was causing our fastlane upload to fail. We have not changed anything from our side regarding app privacy and it seems like others are seeing the same issue as well #18653.

Here is the error message:
`Search query: The request cannot be fulfilled because of the state of another resource. - Submit for review errors found.`
`Answers to what data your app collects and how it's used are needed. - You must have published answers to your app's data usages.`

Any help would be much appreciated thanks!

##### Command executed

`fastlane deliver privacy -f --verbose --api_key_path "****"`

### Environment

<details>
  <pre> <details><summary>✅ fastlane environment ✅</summary>

### Stack
| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.15.1                                     |
| Ruby                        | 2.7.0                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.21.1 (Apple Git-122.3)        |
| Installation Source         | ~/.rvm/gems/ruby-2.7.0/bin/fastlane         |
| Host                        | Mac OS X 10.15.1 (19B88)                    |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.7.0/lib                |
| OpenSSL Version             | OpenSSL 1.1.1g  21 Apr 2020                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 11.3.1                                      |

### System Locale
| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:
<details><summary>`./fastlane/Fastfile`</summary>
ruby
lane :privacy do
  upload_app_privacy_details_to_app_store(
  username: "********",
  app_identifier: "************",
  json_path: "fastlane/app_privacy_details.json",
  team_id: "*********"
)
end
</details>
**No Appfile found**

### fastlane gems
| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.182.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:
**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.5.0        |
| executable-hooks              | 1.6.1        |
| bundler-unload                | 1.0.2        |
| rubygems-bundler              | 1.4.5        |
| bundler                       | 2.2.10       |
| rake                          | 13.0.3       |
| CFPropertyList                | 3.0.3        |
| public_suffix                 | 4.0.6        |
| addressable                   | 2.7.0        |
| artifactory                   | 3.0.15       |
| atomos                        | 0.1.3        |
| aws-eventstream               | 1.1.1        |
| aws-partitions                | 1.446.0      |
| aws-sigv4                     | 1.2.3        |
| jmespath                      | 1.4.0        |
| aws-sdk-core                  | 3.114.0      |
| aws-sdk-kms                   | 1.43.0       |
| aws-sdk-s3                    | 1.93.1       |
| babosa                        | 1.0.4        |
| claide                        | 1.0.3        |
| colored                       | 1.2          |
| colored2                      | 3.1.2        |
| highline                      | 2.0.3        |
| commander                     | 4.6.0        |
| declarative                   | 0.0.20       |
| digest-crc                    | 0.6.1        |
| unf_ext                       | 0.0.7.7      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20190701 |
| dotenv                        | 2.7.6        |
| emoji_regex                   | 3.2.2        |
| excon                         | 0.80.1       |
| faraday-excon                 | 1.1.0        |
| faraday-net_http              | 1.0.1        |
| faraday-net_http_persistent   | 1.1.0        |
| multipart-post                | 2.0.0        |
| ruby2_keywords                | 0.0.4        |
| faraday                       | 1.4.1        |
| http-cookie                   | 1.0.3        |
| faraday-cookie_jar            | 0.0.7        |
| faraday_middleware            | 1.0.0        |
| fastimage                     | 2.2.3        |
| gh_inspector                  | 1.1.3        |
| jwt                           | 2.2.3        |
| memoist                       | 0.16.2       |
| multi_json                    | 1.15.0       |
| os                            | 1.1.1        |
| signet                        | 0.15.0       |
| googleauth                    | 0.16.1       |
| httpclient                    | 2.8.3        |
| mini_mime                     | 1.1.0        |
| trailblazer-option            | 0.1.1        |
| uber                          | 0.1.0        |
| representable                 | 3.1.1        |
| retriable                     | 3.1.2        |
| google-api-client             | 0.38.0       |
| rexml                         | 3.2.5        |
| webrick                       | 1.7.0        |
| google-apis-core              | 0.3.0        |
| google-apis-iamcredentials_v1 | 0.3.0        |
| google-apis-storage_v1        | 0.3.0        |
| google-cloud-env              | 1.5.0        |
| google-cloud-errors           | 1.1.0        |
| google-cloud-core             | 1.6.0        |
| google-cloud-storage          | 1.31.0       |
| json                          | 2.5.1        |
| mini_magick                   | 4.11.0       |
| naturally                     | 2.2.1        |
| plist                         | 3.6.0        |
| rubyzip                       | 2.3.0        |
| security                      | 0.1.3        |
| simctl                        | 1.6.8        |
| terminal-notifier             | 2.0.0        |
| unicode-display_width         | 1.7.0        |
| terminal-table                | 1.8.0        |
| tty-screen                    | 0.8.1        |
| tty-cursor                    | 0.7.1        |
| tty-spinner                   | 0.9.3        |
| word_wrap                     | 1.0.0        |
| nanaimo                       | 0.3.0        |
| xcodeproj                     | 1.19.0       |
| rouge                         | 2.0.7        |
| xcpretty                      | 0.3.0        |
| xcpretty-travis-formatter     | 1.0.1        |
</details>
*generated on:* **2021-05-07**
</details> </pre>
</details>

**Comment by Leonwu21** [comment]
Hi there,

I'm seeing this issue as well - when I specify in my app metadata that it doesn't collect data, the same parameters are not being reflected on App Store Connect.. This used to work up until recently for us.

**Comment by VladlenaSerie** [comment]
Seeing this issue for my apps.

**Comment by joshdholtz** [comment]
Question because I can't seem to replicate this on any of my projects... Do you all have `NSUserTrackingUsageDescription` set?

**Comment by christruong90** [comment]
Where can I find `NSUserTrackingUsageDescription` I dont see this in my codebase

**Comment by joshdholtz** [comment]
@christruong90 This would be in the Info.plist. It was added for iOS 14.5 in December-ish. I was curious if having that affects how the privacy policy stuff works since that is a privacy policy oriented setting 🤷‍♂️

**Comment by CraseWenz1** [comment]
It effectively transfers my application yet it interfaces side we can not choose under forms segments.

[ElonOne](https://elonone.org)

**Comment by christruong90** [comment]
> @christruong90 This would be in the Info.plist. It was added for iOS 14.5 in December-ish. I was curious if having that affects how the privacy policy stuff works since that is a privacy policy oriented setting 🤷‍♂️

Thanks for looking into this @joshdholtz . I tried adding `NSUserTrackingUsageDescription`  to `Info.plist` but there were no changes and its still not updating correctly on appstore connect

**Comment by ceylino** [comment]
@joshdholtz hey, so I set the` NSUserTrackingUsageDescription` field to true in the Info.plist file and it did not actually fix anything. The privacy details json set correctly and the fastlane upload is successful but for some reason the app privacy details on App Store Connect are not being updated :( 

I feel like this could be happening to a lot of people right now but they may not be noticing it as fastlane call succeeds without any errors so they assume everything is updated correctly on the App Store Connect side. 

I am not sure why this is happening now for the past week as it used to work just fine when the app privacy stuff was first released. Could one of the recent fastlane releases have broken something maybe?

Thank you so much for looking into this issue!

**Comment by joshdholtz** [comment]
Hmmm... Nothing in here was touched recently. There was a recent outage/update to App Store Connect and this action uses a private API so my guess is something changed in there 😞

What's weird is this is working fine for me. What state are your apps in? Do you have a new version that is started or not yet? It seems like it might some logic with some state that isn't effecting everything.

**Comment by ceylino** [comment]
It could be, when you say the state of the apps do you mean "Prepare for Submission" and stuff on App Store Connect?

**Comment by Leonwu21** [comment]
> There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.
> 
> Please make sure to update to the latest `fastlane` version and check if that solves the issue. Let us know if that works for you by adding a comment 👍
> 
> Friendly reminder: contributions are always welcome! Check out [CONTRIBUTING.md](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md) for more information on how to help with `fastlane` and feel free to tackle this issue yourself 💪
> 
> This issue will be auto-closed if there is no reply within 1 month.

Confirming this issue is still unresolved after updating to the latest fastlane version (2.185.1)

**Comment by Leonwu21** [comment]
Just an update here:

We've resolved the issue on our end - It turns out we've been calling our Fastfile lanes incorrectly. Previously we've been calling our privacy lane inside our deliver command: "fastlane deliver privacy ..." which gave us a successful response so we assumed everything was working.

We've now made a separate call before the deliver command: "fastlane privacy" which successfully calls our privacy lane and uploads the privacy data to App Store Connect. The "upload_app_privacy_details_to_app_store" function is nested inside the Fastfile lane as documented here: https://docs.fastlane.tools/uploading-app-privacy-details/

@joshdholtz It may be helpful to throw an error in case lanes are referenced inside the deliver command.

Hopefully this helps anyone else!
